### PR TITLE
Make sure all responses are documented

### PR DESCRIPTION
**Pull request by zupo**
Before this fix, non-200 responses did not need to be documented in
openapi spec, because when the view would raise (for example)
`HTTPForbidden` the response validation would be skipped.

With this fix, if you did not document 403 as a response in openapi
spec you would instead get `openapi_core.schema.responses.exceptions.InvalidResponse:
Unknown response http status: 403`.

### CODE REVIEW

**Comment by zupo** [comment]
Forgot to mention: I've run a suite of 343 tests for our pyramid_openapi3 based web app, and they all pass.

File: pyramid_openapi3/__init__.py [hunk 1]
```diff
@@ -35,18 +35,25 @@ def wrapper_view(context, request):
...
-
-                # Validate response and raise if an error is found
-                open_response = PyramidOpenAPIResponse(response)
-                result = settings['response_validator'].validate(request=open_request, response=open_response)
-                result.raise_for_errors()
+            try:
+                if request.openapi_validated.errors and getattr(request.registry, "_pyramid_openapi3_validation_view_name", None):
+                    # handle request errors
+                    view_name = request.registry._pyramid_openapi3_validation_view_name
+                    response = render_view_to_response(context, request, name=view_name, secure=False)
```
Comment (mcdonc) [inline pyramid_openapi3/__init__.py]: Are you sure we want secure=False here?  It's a crazy edge case to register a view via ``config.pyramid_openapi3_validation_error_view()`` that is protected by a permission, but it'd be surprising to me if I did register a view with a permission that got used for this purpose and the permission was ignored.  Leaving it unsecured is probably ok (and even desirable), but we should mention it in the docs; alternately we could omit secure=False and document that if there is a default view permission registered via ``config.set_default_permission`` that the view should be registered as permission=pyramid.security.NO_PERMISSION_REQUIRED (less desirable, I suppose).

File: pyramid_openapi3/__init__.py [hunk 2]
```diff
@@ -35,18 +35,25 @@ def wrapper_view(context, request):
...
+                else:
+                    # Do the view
+                    response = view(context, request)
+
+            except HTTPException as exc:
+                # If view raises one of the HTTPExceptions, catch it
+                # so that we can validate the response and hence make sure
+                # all possible responses are documented in openapi spec.
+                response = exc
```
Comment (mcdonc) [inline pyramid_openapi3/__init__.py]: Makes sense to me.

File: pyramid_openapi3/__init__.py (hunk 1 above)
Comment (zupo) [reply pyramid_openapi3/__init__.py]: Good point, I'll make a note to document it.

**Pull request merged**